plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/BooleanCoercer.java

```diff
@@ -52,4 +55,55 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
             toType.writeSlice(blockBuilder, converted);
         }
     }
+
+    public static class VarcharToBooleanCoercer
+            extends TypeCoercer<VarcharType, BooleanType>
+    {
+        // These values are populated from
+        // https://github.com/apache/hive/blob/4df4d75bf1e16fe0af75aad0b4179c34c07fc975/serde/src/java/org/apache/hadoop/hive/serde2/objectinspector/primitive/PrimitiveObjectInspectorUtils.java#L559
+        private static final Set<String> FALSE_VALUES = ImmutableSet.of(
+                "false",
+                "off",
+                "no",
+                "0",
+                "");
+        private final boolean isOrcFile;
+
+        public VarcharToBooleanCoercer(VarcharType fromType, boolean isOrcFile)
+        {
+            super(fromType, BOOLEAN);
+            this.isOrcFile = isOrcFile;
+        }
+
+        @Override
+        protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int position)
+        {
+            String varcharValue = fromType.getSlice(block, position).toStringUtf8();
+            boolean coercedValue;
+            if (isOrcFile) {
+                try {
+                    // Apache Hive reads 0 as false, numeric string as true and non-numeric string as null for ORC file format
```

Comment:
Could you pls link the source code for this finding? It may be helpful in the maintenance phase.